core/util/src/main/java/edu/cmu/cs/dennisc/codec/BufferUtilities.java

```diff
@@ -190,69 +190,81 @@ public class BufferUtilities {
-    encodeHeader(encoder, buffer, Character.SIZE, buffer.isDirect(), isNativeRequired);
+  public static void encode(BinaryEncoder encoder, CharBuffer buffer) {
+    encodeHeader(encoder, buffer, Character.SIZE);
     while (buffer.hasRemaining()) {
       encoder.encode(buffer.get());
     }
     encoder.flush();
   }
 
-  public static void encode(BinaryEncoder encoder, ShortBuffer buffer, boolean isNativeRequired) {
-    encodeHeader(encoder, buffer, Short.SIZE, buffer.isDirect(), isNativeRequired);
+  public static void encode(BinaryEncoder encoder, ShortBuffer buffer) {
+    encodeHeader(encoder, buffer, Short.SIZE);
     while (buffer.hasRemaining()) {
       encoder.encode(buffer.get());
     }
     encoder.flush();
   }
 
-  public static void encode(BinaryEncoder encoder, IntBuffer buffer, boolean isNativeRequired) {
-    encodeHeader(encoder, buffer, Integer.SIZE, buffer.isDirect(), isNativeRequired);
+  public static void encode(BinaryEncoder encoder, IntBuffer buffer) {
+    encode(encoder, buffer, true);
+  }
+
+  public static void encodeNativeOptional(BinaryEncoder encoder, IntBuffer buffer) {
+    encode(encoder, buffer, false);
+  }
+
+  private static void encode(BinaryEncoder encoder, IntBuffer buffer, boolean isNativeRequired) {
+    encodeHeader(encoder, buffer, Integer.SIZE, isNativeRequired);
     while (buffer.hasRemaining()) {
       encoder.encode(buffer.get());
     }
     encoder.flush();
   }
 
-  public static void encode(BinaryEncoder encoder, LongBuffer buffer, boolean isNativeRequired) {
-    encodeHeader(encoder, buffer, Long.SIZE, buffer.isDirect(), isNativeRequired);
+  public static void encode(BinaryEncoder encoder, LongBuffer buffer) {
+    encodeHeader(encoder, buffer, Long.SIZE);
     while (buffer.hasRemaining()) {
       encoder.encode(buffer.get());
     }
     encoder.flush();
   }
 
-  public static void encode(BinaryEncoder encoder, FloatBuffer buffer, boolean isNativeRequired) {
-    encodeHeader(encoder, buffer, Float.SIZE, buffer.isDirect(), isNativeRequired);
+  public static void encode(BinaryEncoder encoder, FloatBuffer buffer) {
+    encodeHeader(encoder, buffer, Float.SIZE);
     while (buffer.hasRemaining()) {
       encoder.encode(buffer.get());
     }
     encoder.flush();
   }
 
-  public static void encode(BinaryEncoder encoder, DoubleBuffer buffer, boolean isNativeRequired) {
-    encodeHeader(encoder, buffer, Double.SIZE, buffer.isDirect(), isNativeRequired);
+  public static void encode(BinaryEncoder encoder, DoubleBuffer buffer) {
+    encodeHeader(encoder, buffer, Double.SIZE);
     while (buffer.hasRemaining()) {
       encoder.encode(buffer.get());
     }
     encoder.flush();
   }
 
-  public static ByteBuffer decodeByteBuffer(BinaryDecoder decoder, boolean isNativeRequired) {
+  public static ByteBuffer decodeByteBuffer(BinaryDecoder decoder) {
     BufferDetails header = new BufferDetails(decoder);
     ByteBuffer rv = header.createByteBuffer(decoder);
```

Comment:
what is up with this commented block in every function? 🤔

Reply:
All have sat unused for over a decade. I am happy to remove them.

Reply:
Done and pushed.